pex/common.py

```diff
@@ -318,12 +319,16 @@ def cleanup(self):
 
 
 @contextmanager
-def atomic_directory(target_dir, source=None):
+def atomic_directory(target_dir, source=None, exclusive=False):
```

Comment:
Thoughts on not using a default for `exclusive`, so that call sites must decide what to do? This was a subtle bug - I fear future changes won't realize they need to set `exclusive=True`, and won't realize they have a race condition until someone hits in the wild.

Reply:
Done.